docs/data-structures/ceil_in_bst.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: Ceil in Binary Search Tree
+slug: /data-structures/ceil_in_bst
+---
+
+## Introduction
+
+This article will see how to find the ceiling value in a binary search tree(BST). We are given a Binary Search Tree(BST) and an element, and we have to find the ceiling value of that element in BST. In this type of tree(BST), the left child is always smaller than its parent, and the right child is always greater than its parent.
+
+## Explanation 
+
+Ceil means smallest greater or equal value if the element is present, then print that element; otherwise, print the smallest value greater than the given value. If no greater value is present, then print null.
+
+#### Example 1
+
+<img src="https://i.imgur.com/vjoI6QE.jpg?1" width="600" height="350"/>
+
+```
+ Input: x=50,
+ In the above example, we are given a binary search tree(BST) and an element x(50).
+ Root is 44.
+ So given element is greater than root, traverse the right child.
+ Right child is 66 which is greater than given element, so keep this in res variable.
+ Traverse the left child of the node(66).
+ Left child of node(66) is null.
+ So we stop traversing and return res value that is ceil value of 50 that is 66.
+
+```
+## Algorithm
+
+For calculating Ceil in BST, we follow the following algorithm:
```

Comment:
There is No Algorithm present here, so don't write word Algorithm,
 After  or calculating Ceil in BST, write **we follow the following steps** only

Reply:
done